dffml/tuner/random_search.py

```diff
@@ -0,0 +1,127 @@
+        test_data: Union[BaseSource, Record, Dict[str, Any]],
+    ):
+        """
+        Method to optimize hyperparameters by parameter grid.
+        Uses a grid of hyperparameters in the form of a dictionary present in config,
+        Trains each permutation of the grid of parameters and compares accuracy.
+        Sets model to the best parameters and returns highest accuracy.
+
+        Parameters
+        ----------
+        model : ModelContext
+            The Model which needs to be used.
+
+        feature : Feature
+            The Target feature in the data.
+
+        accuracy_scorer: AccuracyContext
+            The accuracy scorer that needs to be used.
+
+        train_data: SourcesContext
+            The train_data to train models on with the hyperparameters provided.
+
+        test_data : SourcesContext
+            The test_data to score against and optimize hyperparameters.
+
+        Returns
+        -------
+        float
+            The highest score value
+        """
+
+        if self.parent.config.objective == "min":
+            highest_acc = float("inf")
+        elif self.parent.config.objective == "max":
+            highest_acc = -1
+        else:
+            raise NotImplementedError('Objective must be either "min" or "max".')
+
+        best_config = dict()
+        logging.info(
+            f"Optimizing model with parameter grid: {self.parent.config.parameters}"
+        )
+
+        names = list(self.parent.config.parameters.keys())
+        logging.info(names)
+
+        with model.parent.config.no_enforce_immutable():
+            for _ in range(self.parent.config.trials):
+                combination = []
+                for pvs in self.parent.config.parameters.values():
+                    combination.append(random.choice(pvs))
+                logging.info(combination)
+
+                for i in range(len(combination)):
+                    param = names[i]
+                    setattr(model.parent.config, names[i], combination[i])
+                await train(model.parent, *train_data)
+                acc = await score(
+                    model.parent, accuracy_scorer, feature, *test_data
+                )
+
+                logging.info(f"Accuracy of the tuned model: {acc}")
+                if self.parent.config.objective == "min":
+                    if acc < highest_acc:
+                        highest_acc = acc
+                        for param in names:
+                            best_config[param] = getattr(
+                                model.parent.config, param
+                            )
+                elif self.parent.config.objective == "max":
+                    if acc > highest_acc:
+                        highest_acc = acc
+                        for param in names:
+                            best_config[param] = getattr(
+                                model.parent.config, param
+                            )
```

Comment:
seems like there could be some code recurring through the different tuners. Let's consider the possibility of using helper/utility functions